buf/registry/module/v1beta1/commit_service.proto

```diff
@@ -17,341 +17,45 @@ syntax = "proto3";
-  // this operation. To associate created Commits, use BranchService.Push or BranchService.Rebase.
-  //
-  // This operation is atomic. Either all Commits and associated content are created or an error is returned.
-  rpc CreateCommits(CreateCommitsRequest) returns (CreateCommitsResponse);
-  // Get the pointers to the content for a given set of Commits, Modules, Branches, Tags, or VCSCommits.
-  //
-  // Nodes consist of:
-  //   - The FileNodes: .proto files, license files, and documentation files.
-  //   - The dependencies.
-  //
-  // Retrieving file content is a multi-step process:
-  //   - Get the FileNode messages via this RPC. FileNodes are a map from path to digest.
-  //   - Get the Blob messages via GetBlobs. Blobs are a map from digest to content.
-  //   - If file content for the dependencies is desired, Get the Commits for the DepNodes,
-  //     and call GetCommitCodes for those dependencies.
-  rpc GetCommitNodes(GetCommitNodesRequest) returns (GetCommitNodesResponse) {
-    option idempotency_level = NO_SIDE_EFFECTS;
-  }
-  // Get the Blobs for files within Commits for a given set of Digests.
-  rpc GetBlobs(GetBlobsRequest) returns (GetBlobsResponse) {
-    option idempotency_level = NO_SIDE_EFFECTS;
-  }
-  // Get the digests for Blobs that are not currently stored on the server.
-  rpc GetMissingBlobDigests(GetMissingBlobDigestsRequest) returns (GetMissingBlobDigestsResponse) {
+  // Get Commits.
+  rpc GetCommits(GetCommitsRequest) returns (GetCommitsResponse) {
     option idempotency_level = NO_SIDE_EFFECTS;
   }
 }
 
-message ResolveCommitsRequest {
+message GetCommitsRequest {
   // References to request a Commit for.
   //
   // See the documentation on ResourceRef for resource resolution details.
   //
-  // Once the resource is resolved, this reference is interpeted to refer to the following Commit:
-  //   - If a Module is referenced, this is interpreted to mean the latest released Commit on the Module.
-  //   - If a Commit is referenced, this just references that specific commit.
-  //   - If a Tag is referenced, this is interpreted to mean the Commit associated with the Tag.
-  //   - If a VCSCommit is referenced, this is interpreted to mean the Commit associated with the VCSCommit.
-  //   - Is a Branch is referenced, this is interpreted to mean the head Commit on the Branch.
-  //   - If a Digest is referenced, this is interpreted to mean the latest released Commit that has this Digest.
-  //     Digests referencing unreleased Commits cannot be referenced.
+  // Resolution is as follows:
+  //   - If a Module is referenced, the Commit of the default Label is returned.
+  //   - If a Label is referenced, the Commit of this Label is returned.
+  //   - If a Commit is referenced, this Commit is returned.
   //
-  // TODO: what is "latest released Commit" if we now have "head Commit"?
+  // TODO: It's a little incongruent to have a Get request take a ResourceRef.
```

Comment:
I actually think the primary use case of a `ResourceRef` _is_ a Get request. `ResourceRef` is really only needed because of the "opaque string" concept. In mutational cases, we should request the most specific thing possible (Commits ID if deps, Label IDs/names if labels to advance, etc.).